File: bank-service/src/main/java/rs/raf/bank_service/domain/dto/CreateClientDto.java
package rs.raf.bank_service.domain.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.*;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor


public class CreateClientDto {
    @NotNull(message = "First name cannot be null")
    @Size(min = 2, max = 100, message = "First name must be between 2 and 100 characters")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "First name must contain only letters")
    private String firstName;

    @NotNull(message = "Last name cannot be null")
    @Size(min = 2, max = 100, message = "Last name must be between 2 and 100 characters")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "First name must contain only letters")
    private String lastName;

    @Past(message = "Date of birth must be in the past")
    private Date birthDate;

    @NotNull(message = "Gender cannot be null")
    @Size(min = 1, max = 1, message = "Gender must be 1 character")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "Gender must be a letter")
    private String gender;

    @NotNull(message = "Email cannot be null")
    @Email(message = "Email should be valid")
    private String email;

    @Pattern(regexp = "^0?[1-9][0-9]{6,14}$", message = "Invalid phone number")
    private String phone;

    @NotNull(message = "Address cannot be null")
    private String address;

    @NotNull(message = "Username cannot be null")
    @Size(min = 2, max = 100, message = "Username must be between 2 and 100 characters")
    private String username;

    @NotNull(message = "Jmbg cannot be null")
    @Pattern(
            regexp = "^[0-9]{13}$",
            message = "Jmbg must be exactly 13 digits"
    )
    private String jmbg;
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/dto/client/CreateClientDto.kt
package pack.userservicekotlin.domain.dto.client

import jakarta.validation.constraints.*
import jakarta.validation.constraints.Past
import java.util.*

data class CreateClientDto(
    @NotNull(message = "First name cannot be null")
    @Size(min = 2, max = 100, message = "First name must be between 2 and 100 characters")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "First name must contain only letters") val firstName: String? = null,
    @NotNull(message = "Last name cannot be null")
    @Size(min = 2, max = 100, message = "Last name must be between 2 and 100 characters")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "First name must contain only letters") val lastName: String? = null,
    val birthDate:
        @Past(message = "Date of birth must be in the past")
        Date? = null,
    @NotNull(message = "Gender cannot be null")
    @Size(min = 1, max = 1, message = "Gender must be 1 character")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "Gender must be a letter") val gender: String? = null,
    @NotNull(message = "Email cannot be null")
    @Email(message = "Email should be valid") val email: String? = null,
    @Pattern(regexp = "^0?[1-9][0-9]{6,14}$", message = "Invalid phone number") val phone: String? = null,
    @NotNull(message = "Address cannot be null") val address: String? = null,
    @NotNull(message = "Username cannot be null")
    @Size(min = 2, max = 100, message = "Username must be between 2 and 100 characters") val username: String? = null,
    @NotNull(message = "Jmbg cannot be null")
    @Pattern(regexp = "^[0-9]{13}$", message = "Jmbg must be exactly 13 digits") val jmbg: String? = null,
)
